modules/battery_manager/ajax/add_entry.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Battery Manager adder
+ *
+ * Checks Test Battery for duplicates and handles insertions into Test Battery
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+
+// Determine which action is called
+if (isset($_GET['action'])) {
+    $action = $_GET['action'];
+    if ($action == "checkForDuplicate") {
+        echo checkForDuplicate();
+    } else if ($action == "add") {
+        addEntry();
+    }
+}
+
+/**
+ * Return duplicate of entry if it exists and null otherwise
+ *
+ * @return Object or null
+ */
+function checkForDuplicate()
+{
+    $db =& \Database::singleton();
+
+    // Retrieve values entered by user
+    $form_data = getFormData();
+
+    // Build SQL query based on values entered by user
+    $query     = " SELECT * FROM test_battery ";
+    $i         = 0;
+    $connector = "WHERE ";
+    foreach ($form_data as $key => $value) {
+        if ($i > 0) {
+            $connector = "AND ";
+        }
+        if (isset($value)) {
+            $query .= $connector . $key . " = :" . $key . " ";
+        } else {
+            $query .= $connector . $key . " IS NULL ";
```

Comment:
if a field is empty this would return false and yet they would be functionally equivalent, you should add a check for the empty case